docs/user_guide/datasets/boston_housing_data.rst

```diff
@@ -1,6 +1,4 @@
-Datasets
-==========
-
+.. _boston-housing-data:
```

Comment:
I don't know if dashes work in these tags, but let's stick to underscores.
```suggestion
.. _boston_housing_data:
```

Reply:
My understanding was that tags for cross-referencing should use dashes, or at least that's what the documentation uses as an example. Please help me learn if I'm incorrect https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#cross-referencing-arbitrary-locations

Reply:
Interesting. Not sure why I thought this to be true then....

Since we're doing it this way everywhere else I'd still suggest sticking with underscore, though. If we ever decide to go to dashes we should do it everywhere.

Reply:
Makes sense. I just updated everything to use underscores